vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SnapshotManager.java

```diff
@@ -0,0 +1,370 @@
+import edu.asu.diging.vspace.core.exception.SequenceNotFoundException;
+import edu.asu.diging.vspace.core.exception.SlideNotFoundException;
+import edu.asu.diging.vspace.core.exception.SlidesInSequenceNotFoundException;
+import edu.asu.diging.vspace.core.file.IStorageEngine;
+import edu.asu.diging.vspace.core.file.IStorageManager;
+import edu.asu.diging.vspace.core.model.IChoice;
+import edu.asu.diging.vspace.core.model.IContentBlock;
+import edu.asu.diging.vspace.core.model.IExhibition;
+import edu.asu.diging.vspace.core.model.IModule;
+import edu.asu.diging.vspace.core.model.IModuleLink;
+import edu.asu.diging.vspace.core.model.ISequence;
+import edu.asu.diging.vspace.core.model.ISlide;
+import edu.asu.diging.vspace.core.model.ISpace;
+import edu.asu.diging.vspace.core.model.IVSImage;
+import edu.asu.diging.vspace.core.model.display.ISpaceLinkDisplay;
+import edu.asu.diging.vspace.core.model.impl.BranchingPoint;
+import edu.asu.diging.vspace.core.model.impl.ExhibitionDownload;
+import edu.asu.diging.vspace.core.model.impl.SequenceHistory;
+import edu.asu.diging.vspace.core.model.impl.SnapshotTask;
+import edu.asu.diging.vspace.core.model.impl.Space;
+import edu.asu.diging.vspace.core.model.impl.SpaceStatus;
+import edu.asu.diging.vspace.core.services.IExhibitionManager;
+import edu.asu.diging.vspace.core.services.IExternalLinkManager;
+import edu.asu.diging.vspace.core.services.IModuleLinkManager;
+import edu.asu.diging.vspace.core.services.IModuleManager;
+import edu.asu.diging.vspace.core.services.ISequenceManager;
+import edu.asu.diging.vspace.core.services.ISnapshotManager;
+import edu.asu.diging.vspace.core.services.ISpaceDisplayManager;
+import edu.asu.diging.vspace.core.services.ISpaceLinkManager;
+import edu.asu.diging.vspace.core.services.ISpaceManager;
+
+@Service
+public class SnapshotManager implements ISnapshotManager {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    
+    @Autowired
+    private SpaceRepository spaceRepository;
+    
+    @Autowired
+    private IModuleManager moduleManager;
+
+    @Autowired
+    private SlideManager slideManager;
+
+    @Autowired
+    private ISequenceManager sequenceManager;
+    
+    @Autowired
+    @Qualifier("storageEngineDownloads")
+    private IStorageEngine storageEngineDownloads;
+    
+    @Autowired
+    private SpringTemplateEngine springTemplateEngine;
+
+    @Autowired
+    private ISpaceDisplayManager spaceDisplayManager;
+
+    @Autowired
+    private IExhibitionManager exhibitManager;
+
+    @Autowired
+    private IModuleLinkManager moduleLinkManager;
+
+    @Autowired
+    private ISpaceLinkManager spaceLinkManager;
+
+    @Autowired
+    private IExternalLinkManager externalLinkManager;
+    
+    @Autowired
+    private ISpaceManager spaceManager;
+    
+    @Autowired
+    private SnapshotTaskRepository snapshotTaskRepository;
+    
+    @Autowired
+    private IStorageManager storageManager;
```

Comment:
it looks like this class is only used here and has only one method. I don't think you need it. Just move the method here since there is already a dependency on one of the storage engines.

Reply:
@jdamerow , This class was created following your comment that if we want to move anything from uploads to downloads, a separate manager class should handle it.